linkerd/meshtls/rustls/src/client.rs

```diff
@@ -93,7 +92,8 @@ where
 
     fn call(&mut self, io: I) -> Self::Future {
         tokio_rustls::TlsConnector::from(self.config.clone())
-            .connect(self.server_id.as_ref(), io)
+            // XXX(eliza): it's a bummer that the server name has to be cloned here...
+            .connect(self.server_id.clone(), io)
```

Comment:
this is not my favorite thing about the new `rustls` API, but i don't think we can fix it in the proxy...